### ISSUE TITLE
Run script commands inside WeeChat buffer

### ISSUE THREAD

**Issue by GermainZ** [weechat/weechat#128]
As discussed in #weechat:

```
GermainZ | Can we run an interactive Python shell inside WeeChat?
[…]
@FlashCode | btw, such feature could be added in scripting plugins (or in script plugin)
@FlashCode | to have it in all languages
@FlashCode | and a way to run some python/perl/... code on the command line too
@FlashCode | (without the buffer I mean)
```

Example usage:
Open a new Python shell buffer in WeeChat: `/python shell`
Execute command and show output directly (assuming weechat is automatically imported): `/python shell weechat.info_get("version_number", '')`
(similarly for other languages)

**Comment by flashcode** [comment]
Implemented in Python, Perl, Ruby, Lua and Guile plugins (version 2.1).

Still missing in TCL, Javascript and PHP plugins; this will be implemented in an upcoming release.
For these plugins, we must first find a way to redirect script stdout/stderr, help is welcome for this.

**Comment by flashcode** [comment]
Partially done in version 2.1, removing the milestone for now.

**Comment by flashcode** [comment]
For you info, this feature was removed from Guile plugin in version 2.7, because now that WeeChat is compatible with Guile 2.2, this feature was broken and not so easy to repair, so I removed it.
It could be reintroduced in near future if I find a clean fix.

**Comment by reyesjl** [comment]
this still up for grabs? would love to contribute something here

**Comment by flashcode** [comment]
@reyesjl: yes, currently the eval feature is working in these scripting languages:

* Python
* Perl
* Ruby
* Lua
* Guile

It's still missing in these languages, and help is welcome:

* Javascript
* PHP
* TCL

If you want to help, the function `weechat_xxx_eval` must be completed in the main C language file for the plugin:

* Javascript: https://github.com/weechat/weechat/blob/master/src/plugins/javascript/weechat-js.cpp#L553-L572
* PHP: https://github.com/weechat/weechat/blob/master/src/plugins/php/weechat-php.c#L875-L894
* TCL: https://github.com/weechat/weechat/blob/master/src/plugins/tcl/weechat-tcl.c#L557-L576

**Comment by reyesjl** [comment]
Great news – I will have a look at this now for an hour or so.

Thank you,
Jose

**Comment by reyesjl** [comment]
all done check it out #1850
